File: Java Spring Boot Project/CompanyDirectory/src/main/java/com/mindthecode/CompanyDirectory/models/responses/BusinessUnitResponse.java
package com.mindthecode.CompanyDirectory.models.responses;

import com.mindthecode.CompanyDirectory.models.entities.Company;

public class BusinessUnitResponse {

    private long id;
    private String name;
    private Company company;

    public BusinessUnitResponse() {
    }

    public BusinessUnitResponse(long id, String name, Company company) {
        this.id = id;
        this.name = name;
        this.company = company;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Company getCompany() {
        return company;
    }

    public void setCompany(Company company) {
        this.company = company;
    }

    @Override
    public String toString() {
        return "BusinessUnitResponse{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", company=" + company +
                '}';
    }
}

File: Java Spring Boot Project/company_directory_app/lib/models/responses/business_unit_response.dart
import 'package:company_directory_app/models/entities/company.dart';

class BusinessUnitResponse {
  double id;
  String name;
  Company company;

  BusinessUnitResponse({this.id, this.name, this.company});

  factory BusinessUnitResponse.fromJson(Map<String, dynamic> json) {
    return BusinessUnitResponse(
      id: json['id'],
      name: json['name'],
      company: json['company'],
    );
  }
}
